app/code/Magento/CatalogUrlRewrite/Model/Category/Plugin/Storage.php

```diff
@@ -59,6 +59,7 @@ public function afterReplace(StorageInterface $object, array $result, array $url
             ];
         }
         if (count($toSave) > 0) {
+            $this->productResource->removeMultiple(array_column($toSave, 'url_rewrite_id'));
             $this->productResource->saveMultiple($toSave);
```

Comment:
I just looked at the implementation of the saveMultiple method uses 
https://github.com/magento/magento2/blob/ef9425dd5154fa64f8d0de570e5644992b905213/app/code/Magento/CatalogUrlRewrite/Model/ResourceModel/Category/Product.php#L56

What if we'll replace it with the `insertOnDuplicate` method, which should handle duplicates correctly?

Reply:
Yes, that would be essentially the same as saveMultiple is only called from the place that I changed.
But as it is a public method, it might eventually be called from other places and it would change behavior slightly, but I guess that is not actually a problem

Reply:
Shall I change it ?

Reply:
Other option would be to change the signature to

    public function saveMultiple(array $insertData, $ignoreDuplicates = false)

Reply:
I think no need to add an optional parameter as the exception here isn't something expected, so better to fix the implementation. In most cases, it works fine, but in some edge cases, it works incorrectly.

PS: it's not used anywhere else according to https://github.com/magento/magento2/search?q=saveMultiple

Reply:
Change was made